### PR TITLE
Modal com_categories: improve Select & Edit Category

### PR DESCRIPTION
**Pull request by cyrez**
Improve modals to select and to edit a category.
**Note: to be testing on staging.**
#### Summary of Changes
- When Bootstrap modal loads an iframe, BS tooltips with placement top are truncated at the top border of the iframe. (no auto placement in BS2). This PR sets the tooltip placement to bottom to fix this issue.
- Remove inline CSS
- Content is embedded in a container div (using already existing container-popup class for modal.php)
- Add empty lines to improve the readability
- Add viewport dimensions (new <code>3.6.0-dev</code>)
- Category edit modal is simplify (loading edit layout, so no dupplicated code)
- Fix Edit button when current associated category in data is not the category selected
- Control if modal, in edit layout
- Add Save, Save & Close and Close button to the footer of the modal
- Fix Edit modal not opening as a modal (missing bootstrap.renderModal for Edit modal)
- Fix <code>Uncaught SyntaxError: Unexpected token )</code> in select category modal
#### Testing Instructions (Staging + Multilanguages enabled!)
- Go to Contents > Categories > open a categorie, and go to <code>Associations</code> tab to select associated category to open the modal.
- Verify tooltip placement is bottom, and not truncated
- Verify the modal size now is proportionnal to viewport (width 80/100th viewport width, and modal-body is 70/100th viewport height)
- Select an associated category and save the category to show edit button
- Open Edit modal and play with it ;-)

**Select Category Before Patch**
capture d ecran 2016-05-12 a 16 43 11

**Select Category After Patch**
capture d ecran 2016-05-12 a 16 42 16

**Edit Category Before Patch**
Open in a new window...
capture d ecran 2016-05-13 a 18 57 41

**Edit Category After Patch**
Open in a modal
capture d ecran 2016-05-13 a 18 58 39

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
in the modal, got a JS error `Uncaught SyntaxError: Unexpected token )` on selecting a category.

BTW did you notice the lack of modal in the "Edit" button (the button appears after saving the category with the association)?

**Comment by andrepereiradasilva** [comment]
ups, the js error is already there without your PR ..., so tested with success

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on d738268a8a5dda5a3919dc6e358731d0f390cf55<br /><br />as said js error was already there so this PR does what is proposed to do.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10441">issues.joomla.org/joomla-cms/10441</a>.</sub>

**Comment by JoomliC** [comment]
@andrepereiradasilva yes error was there before, but prepare yourself for a new test, as i will fix it (this PR is a good occasion to fix issue!) ;-)

**Comment by JoomliC** [comment]
@andrepereiradasilva JS error is now fixed here : https://github.com/joomla/joomla-cms/pull/10441/commits/96e2bb0b5b0c84b69be6a2791f03b40dc4996209
But, i will put this PR in stand-by until tomorrow, as i will see to add directly in this PR, the modal to edit category and do a 2-in-1 PR for modals select and edit category ;-)

**Comment by andrepereiradasilva** [comment]
ok will wait then.

**Comment by JoomliC** [comment]
@andrepereiradasilva That's ok!
- Issue Fixed
- And PR for both Select and Edit category modals ;-)

Ready for testing! (and to be then the base for other ones, if all is ok!) 👍

File: administrator/components/com_categories/models/fields/modal/category.php [hunk 1]
```diff
@@ -137,11 +144,19 @@ protected function getInput()
 			$value = (int) $this->value;
 		}
 
+		$urlSelect = $linkCategories . '&amp;' . JSession::getFormToken() . '=1';
+		$urlEdit   = $linkCategory . '&amp;id=' . $value . '&amp;' . JSession::getFormToken() . '=1';
```
Comment (andrepereiradasilva) [inline administrator/components/com_categories/models/fields/modal/category.php]: shouldn't all internal URL be passed trough `JRoute::_()`?

Comment (JoomliC) [inline administrator/components/com_categories/models/fields/modal/category.php]: The urls there are the source url (remote) for the iframe inside the modal.
So, as inside the modal-body iframe of the modal, should not be needed, and not sure in fact that is should be used... (none of all bootstrap modals iframe use JRoute ;-) )

Comment (andrepereiradasilva) [inline administrator/components/com_categories/models/fields/modal/category.php]: humm ... ok then i guess is not needed here

Comment (JoomliC) [inline administrator/components/com_categories/models/fields/modal/category.php]: Indeed, this is a good question... and wonder who can answer ? :-|

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 443897d8c1b053bb20ba6d2d94e034aef84757db<br /><br />Tested with success.

Solves a lot of issues with the "Edit" modal, js errors and the "Clear" button.

Also now the "Edit" is a modal with "Close", "Save" and "Save & Close" buttons! :+1:<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10441">issues.joomla.org/joomla-cms/10441</a>.</sub>

**Comment by andrepereiradasilva** [comment]
@JoomliC really nice work!

The modals are getting much more user friendly.
Also you're solving a lot of bugs!

I hope all modals get reviewed for 3.6.0.

**Comment by JoomliC** [comment]
Thank you @andrepereiradasilva !
Yes, if this one for category is ok, i will update article and module (with save button) and will do the ones for contact and newsfeed.
I think i can do this for 3.6.0 ;-)

**Comment by BurtNL** [comment]
I have tested this item :white_check_mark: successfully on 443897d8c1b053bb20ba6d2d94e034aef84757db<br /><br />Everything works as described. Great!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10441">issues.joomla.org/joomla-cms/10441</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10441">issues.joomla.org/joomla-cms/10441</a>.</sub>

**Comment by JoomliC** [comment]
@andrepereiradasilva @BurtNL Thanks for testing!
A new one just added for com_contacts #10457 ;-)

**Pull request merged**